src/unix/mod.rs

```diff
@@ -135,7 +135,7 @@ cfg_if! {
     if #[cfg(not(stdbuild))] {
         // cargo build, don't pull in anything extra as the libstd  dep
         // already pulls in all libs.
-    } else if #[cfg(target_env = "musl")] {
+    } else if #[cfg(all(target_env = "musl", target_arch = "x86_64"))] {
```

Comment:
From my point of view, x86_64-musl is the outlier here because it's the only target that links to libc statically :P.